David/pdfreports

DavidSarlle opened the pull request:
This PR implements:

- Automatic resizing of plotting area when multiple figures are returned in a plotting function
- Added fitting report in a styled html table instead of using hover property of plotly link
- Restyled fitting area and refresh rate area in plots 
- Added funcionality for exporting into pdf file or print any report
- Minor fixes intemplate html styling  

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

DavidSarlle commented:
> @DavidSarlle I was looking into this, may I ask what is the motivation to generate PDF reports given that qibocal provides tools to store reports and share URL links?

@scarrazza Some of the advantages are:
1. It gives the users the possibilty of exporting/saving/print reports directly from the qibocal GUI
2. Adds extra funcionlity for sharing to qibocal
3. The funcionality gives a user-frinedly option for sharing reports, without using qq-export
4. With this funcionality, the user can generate reports even if the calibration routine has no finished (useful if the routine crasesh before ending)
5. Report integration into larger reports can be done more easily with pdf documents than with URLs

These are the reasons that have motivated the implementation of this functionality. Talking with fabrication team and experimentalist at TII, they were happy with this possibility.

andrea-pasquale commented on src/qibocal/web/app.py:
```diff
@@ -10,6 +11,9 @@
 from qibocal.data import DataUnits
 from qibocal.web.server import server
 
+# import pandas as pd
+# from collections import OrderedDict
+
```
We can remove this if they are no longer needed.

DavidSarlle replied:
Done

andrea-pasquale commented on src/qibocal/web/app.py:
```diff
@@ -117,16 +113,60 @@ def get_graph(interval, url, value):
+            table_rows = []
+
+            for i in range(0, len(fitting_params), 3):
+                # print(fitting_params[i])
```
This line can also be removed.

DavidSarlle replied:
Done

andrea-pasquale reviewed (commented):
Thanks @DavidSarlle.
I like the table that you implemented to display the results from the fitting.
I think that the pdf report is not working as it should:
![image](https://user-images.githubusercontent.com/49183315/218962307-4c165283-d250-4cb4-b97f-a431e9995914.png)
Can you reproduce this error?

andrea-pasquale commented:
I see also that this PR is linked to #229, however I don't think that this issue is addressed in this PR, correct?

DavidSarlle commented:
> I see also that this PR is linked to #229, however I don't think that this issue is addressed in this PR, correct?

It was somehow related because the possibility to export or print at any point of the execution of a calibration routine could be exported at that point. So, the user has the possibility to print/save an intermediate report. But, for sure is not automatic. the implementation in #245 is the best option.

DavidSarlle commented:
@andrea-pasquale thanks for the comment. I've been able to reproduce the problem that you pointed out. I have modified the css @media print, so all the elements in the report fit in a A4 portarit orientation (and, of course in a A4 landscape). Fitting the iframes that allocates the content from the dask app (the figures and the fitting data) for all resolutions is not easy. I'have done my best, and now the reports are good enought for almost all resolutions. 
I have also elimnated some info from the pdf report to be able to show more than one figures per page. basically I have removed the summary. See below examples for typical resolutions:

width: 1920px
![image](https://user-images.githubusercontent.com/39095664/219612646-e009033b-dab8-41c0-9a49-e581b0e70fc1.png)

width: 1366px
![image](https://user-images.githubusercontent.com/39095664/219612542-483a4f8e-1aff-428f-a783-c0060ed78d8c.png)

DavidSarlle commented:
@andrea-pasquale @scarrazza the PR can also be included in the first release if you are happy with the implementation. The most importat thing included in this PR is the auto resizing of the plotting area and the way of showing the fitting data. I think that is interersting if we include the PR in the first release.

scarrazza commented:
Looks good to me, @andrea-pasquale let me know.

andrea-pasquale commented:
Thanks @DavidSarlle. LGTM now.

andrea-pasquale commented:
@scarrazza when can merge it now. Tests are failing only for coverage of the plottng methods which currently we don't test.
